src/metabase/api/macros.clj

```diff
@@ -129,15 +129,42 @@
             (s/spec (s/cat :path    string?
                            :regexes (s/+ ::defendpoint.route.key-regex-pair))))))
 
+(mu/defn- auto-route-param-regex :- [:maybe (ms/InstanceOfClass java.util.regex.Pattern)]
+  [route-param :- :keyword
+   route-params-schema]
+  (when (and route-params-schema
+             (= (mc/type route-params-schema) :map))
+    (some (fn [[k _options v-schema]]
+            (when (= k route-param)
+              (or
+               (second (metabase.api.common.internal/->matching-regex v-schema))
+               (:api/regex (mc/properties v-schema)))))
+          (mc/children route-params-schema))))
+
+(mu/defn- auto-route-regexes :- [:maybe [:map-of :keyword (ms/InstanceOfClass java.util.regex.Pattern)]]
+  "Auto-infer regexes for the route based on the route args schema. These are either defined
+  in [[metabase.api.common.internal/->matching-regex]] or defined in the schema itself with the `:api/regex` key."
```

Comment:
Worth updating this to explain the precidence for resolving the regex to use.

something like:

> These are resolved by looking for the `:api/regex` key on the schema itself, and if that is missing, [[metabase.api.common.internal/->matching-regex]].

Should we warn if there's no regex generated for a schema? In defendpoint1 there is a set of excluded schemas for that purpose: https://github.com/metabase/metabase/blob/master/src/metabase/api/common/internal.clj#L246

Reply:
good idea, I've added this

```
2025-01-24 23:40:37,682 WARN api.macros :: Warning: no regex explicitly specified or inferred for metabase.api.search :get "/:fish" route param :fish, add an :api/regex key to its schema or explicitly specify route regexes.
```